src/System/include/BipedalLocomotion/System/VariablesHandler.h

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file VariablesHandler.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <string>
+#include <unordered_map>
+
+#include <iDynTree/Core/Utils.h>
+
+#ifndef BIPEDAL_LOCOMOTION_SYSTEM_ADVANCEABLE_VARIABLES_HANDLER_H
+#define BIPEDAL_LOCOMOTION_SYSTEM_ADVANCEABLE_VARIABLES_HANDLER_H
+
+namespace BipedalLocomotion
+{
+namespace System
+{
+/**
+ * VariableHandler is useful to handle variables in an optimization problem, Their name, dimension
+ * and position
+ */
+class VariablesHandler
+{
+
+    std::unordered_map<std::string, iDynTree::IndexRange> m_variables; /**< Map containing the name
+                                                                          of a variable and its
+                                                                          index range */
+    std::size_t m_numberOfVariables{0}; /**< Total number of Variable seen as scalar */
+
+public:
+    /**
+     * Add a mew variable to the list
```

Comment:
```suggestion
     * Add a new variable to the list
```
There are a few similar typos.